ramalama/model.py

```diff
@@ -180,6 +178,20 @@ def cleanup():
         run_cmd(conman_args, stdout=None, debug=args.debug)
         return True
 
+    def gpu_args(self):
+        gpu_args = [ ]
+        if sys.platform == "darwin":
+            # llama.cpp will default to the Metal backend on macOS, so we don't need
+            # any additional arguments.
+            pass
+        elif sys.platform == "linux" and (os.path.exists("/dev/dri") or
+              os.getenv("HIP_VISIBLE_DEVICES") or os.getenv("CUDA_VISIBLE_DEVICES")):
+            gpu_args = ["-ngl", "99"]
+        else:
+            print("GPU offload was requested but is not available on this system")
```

Comment:
I don't think there's a "right" answer to that question. It's mostly a UX decision. Perhaps it'll be less confusing if we stop the application so the users don't miss the message but, on the other hand, llama.cpp's behavior is to ignore `-ngl 99` if there's any kind of problem trying to offload the model to the GPU, so there'll be still plenty of cases in which this may happen.